From 4417d8266afe3ef4cf3838ffbd4df80ab7dbc850 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Fri, 27 Jan 2012 16:36:07 +0100
Subject: [PATCH 16/22] usb-msc: move cdb dispatch to block.c

RH-Author: Paolo Bonzini <pbonzini@redhat.com>
Message-id: <1327682173-30723-17-git-send-email-pbonzini@redhat.com>
Patchwork-id: 36980
O-Subject: [RHEL6.3 PATCH seabios 16/22] usb-msc: move cdb dispatch to block.c
Bugzilla: 782028
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Alon Levy <alevy@redhat.com>

virtio-scsi's low-level dispatch code is exactly the same as USB's,
since in the end both are actually SCSI HBAs.  Move it to common code.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry-picked from 1e749c8583a5a8cf7a08f0e0faa2cb1edc3e44c7)

Conflicts:
	src/block.c (no AHCI)
---
 src/block.c   |   28 +++++++++++++++++++++++++---
 src/usb-msc.c |   28 ----------------------------
 2 files changed, 25 insertions(+), 31 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 src/block.c   |   28 +++++++++++++++++++++++++---
 src/usb-msc.c |   28 ----------------------------
 2 files changed, 25 insertions(+), 31 deletions(-)

diff --git a/src/block.c b/src/block.c
index 72e7081..c830259 100644
--- a/src/block.c
+++ b/src/block.c
@@ -10,8 +10,8 @@
 #include "cmos.h" // inb_cmos
 #include "util.h" // dprintf
 #include "ata.h" // process_ata_op
-#include "usb-msc.h" // process_usb_op
 #include "virtio-blk.h" // process_virtio_op
+#include "blockcmd.h" // cdb_*
 
 u8 FloppyCount VAR16VISIBLE;
 u8 CDCount;
@@ -261,6 +261,28 @@ map_floppy_drive(struct drive_s *drive_g)
  * 16bit calling interface
  ****************************************************************/
 
+int
+process_scsi_op(struct disk_op_s *op)
+{
+    if (!CONFIG_USB_MSC)
+        return 0;
+    switch (op->command) {
+    case CMD_READ:
+        return cdb_read(op);
+    case CMD_WRITE:
+        return cdb_write(op);
+    case CMD_FORMAT:
+    case CMD_RESET:
+    case CMD_ISREADY:
+    case CMD_VERIFY:
+    case CMD_SEEK:
+        return DISK_RET_SUCCESS;
+    default:
+        op->count = 0;
+        return DISK_RET_EPARAM;
+    }
+}
+
 // Execute a disk_op request.
 int
 process_op(struct disk_op_s *op)
@@ -278,10 +300,10 @@ process_op(struct disk_op_s *op)
         return process_ramdisk_op(op);
     case DTYPE_CDEMU:
         return process_cdemu_op(op);
-    case DTYPE_USB:
-        return process_usb_op(op);
     case DTYPE_VIRTIO:
 	return process_virtio_op(op);
+    case DTYPE_USB:
+        return process_scsi_op(op);
     default:
         op->count = 0;
         return DISK_RET_EPARAM;
diff --git a/src/usb-msc.c b/src/usb-msc.c
index 72ad15b..cde08ce 100644
--- a/src/usb-msc.c
+++ b/src/usb-msc.c
@@ -116,34 +116,6 @@ fail:
 
 
 /****************************************************************
- * Drive ops
- ****************************************************************/
-
-// 16bit command demuxer for ATAPI cdroms.
-int
-process_usb_op(struct disk_op_s *op)
-{
-    if (!CONFIG_USB_MSC)
-        return 0;
-    switch (op->command) {
-    case CMD_READ:
-        return cdb_read(op);
-    case CMD_WRITE:
-        return cdb_write(op);
-    case CMD_FORMAT:
-    case CMD_RESET:
-    case CMD_ISREADY:
-    case CMD_VERIFY:
-    case CMD_SEEK:
-        return DISK_RET_SUCCESS;
-    default:
-        op->count = 0;
-        return DISK_RET_EPARAM;
-    }
-}
-
-
-/****************************************************************
  * Setup
  ****************************************************************/
 
-- 
1.7.7.6

